libfoundation/src/foundation-string.cpp

```diff
@@ -350,7 +350,10 @@ bool MCStringCreateWithBytes(const byte_t *p_bytes, uindex_t p_byte_count, MCStr
 				}
 				else
 				{
-					// Split to surrogate pairs
+                    // Split to surrogate pairs
+                    // SN-2015-07-03: [[ Bug 15571 ]] Creating a surrogate pair
+                    //  makes the UTF-16 string longer.
+                    t_buffer . Extend(t_buffer . Size() + 1);
```

Comment:
This can fail due to OOM - check the return value and return false if necessary.